dali/pipeline/operators/decoder/nvjpeg_decoder_cpu.h

```diff
@@ -181,8 +192,12 @@ class nvJPEGDecoderCPUStage : public Operator<CPUBackend> {
           NVJPEG_CALL(nvjpegJpegStateDestroy(s->decoder_hybrid_state));
       });
 
-      // We want to use nvJPEG default pinned allocator
-      NVJPEG_CALL(nvjpegBufferPinnedCreate(handle_, nullptr, &state_p->pinned_buffer));
+      {
+        std::lock_guard<std::mutex> l(pool_mutex_);
```

Comment:
Do we need that mutex in dtor?

Reply:
I dont think so, because the executor ensure the worker thread Run finished before calling the op dtor

Reply:
It's not really in dtor...

Reply:
My bad

Reply:
Oh right, I thought you anted to add the lock in the dtor too.